examples/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.21)
+include($ENV{XMOS_CMAKE_PATH}/xcommon.cmake)
+project(lib_adat)
+
+# Get a list of all subdirectories in the current directory
+file(GLOB subdirs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} */)
```

Comment:
In the repos I've been converting to XCommon CMake, I have been explicitly listing all the `add_subdirectory` commands instead of globbing like this. Did you do this so that it automatically finds new examples without you having to add them? That seems useful, but con is this is more complicated. But I think I like it this way.

Reply:
After I did this I re-considered and think a simple add_subdirectory is better. If we're worried about missing examples we should add some test to check for them